src/editor/editor.ts

```diff
@@ -182,23 +185,28 @@ export class Editor {
 
   public updateToolbox(shownPythonToolboxCategories: Set<string>): void {
     if (this.currentModule) {
-      if (this.currentModule.moduleType === commonStorage.MODULE_TYPE_PROJECT) {
-        // If we are editing a Project, we don't add any additional blocks to
-        // the toolbox.
-        this.setToolbox(getToolboxJSON([], shownPythonToolboxCategories));
-        return;
-      }
-      // Otherwise, we add the exported blocks from the Project.
       if (!this.projectContent) {
         // The Project content hasn't been fetched yet. Try again in a bit.
         setTimeout(() => {
           this.updateToolbox(shownPythonToolboxCategories)
         }, 50);
         return;
       }
+      switch(this.currentModule.moduleType){
+        case commonStorage.MODULE_TYPE_PROJECT:
+          this.setToolbox(toolboxRobot.getToolboxJSON(shownPythonToolboxCategories));
+          break;
+        case commonStorage.MODULE_TYPE_MECHANISM:
+          this.setToolbox(toolboxMechanism.getToolboxJSON(shownPythonToolboxCategories));
+          break;
+        case commonStorage.MODULE_TYPE_OPMODE:
+/*
```

Comment:
Add a comment saying why this is commented out. Something like...
TODO: When editing an opmode, the toolbox should show blocks for calling functions that the user has defined in the robot module. This commented-out code will probably need to be reworked.